Fix `Cargo.toml`

jyhi opened the pull request:
This PR tries to improve the quality of `Cargo.toml`s.

Fix #744.

jyhi committed:
**/Cargo.toml,book.toml: change authors to IOTA Stiftung (#744)

jyhi committed:
**/Cargo.toml: fix metadata

Unify names, descriptions, etc.

Thoralf-M commented on bindings/nodejs/native/Cargo.toml:
```diff
@@ -1,12 +1,12 @@
-
 [package]
-name = "iota-client"
+name = "iota-client-nodejs"
```
Here it's the same issue as for wallet.rs `The crate name will be read from the npm_package_name environment variable. If the package name includes a namespace (@namespace/package), the namespace will be removed when matching the crate name (package).`

jyhi replied:
So it's `cargo-cp-artifact` having such behavior. iota.rs does not use it during the Node.js / WASM bindings build. Will it be added in the future?

jyhi replied:
I'm fixing this anyways, we don't publish the crates.

Thoralf-M commented on bindings/wasm/native/Cargo.toml:
```diff
@@ -1,16 +1,12 @@
-
 [package]
-name = "client-wasm"
+name = "iota-client-wasm"
```
I think here it's also related to the npm package name, but needs to be verified

jyhi committed:
bindings/{nodejs,wasm}: fix crate names

cargo-cp-artifact reads the crate name and matches it with the one
specified in `package.json`, so they must be the same.

Thoralf-M commented on Cargo.toml:
```diff
@@ -1,21 +1,12 @@
-readme = "README.md"
+description = "The official, general-purpose IOTA client library in Rust for interaction with the IOTA network (Tangle)"
 repository = "https://github.com/iotaledger/iota.rs"
-homepage = "https://www.iota.org"
```
Why do you want to remove this?

jyhi replied:
I think it's some copy-pasta mistakes -- wallet doesn't have one either.

jyhi committed:
**/Cargo.toml: add more entries

- homepage
- documentation, to wiki
- categories
